### PR TITLE
initialise diff chapter

### PR DESCRIPTION
**Pull request by hristog**
Summary:
- Initialises the `diff` chapter.
- Proposes an outline of the chapter.
- Discusses what's in scope (subject to time constraints, in terms of reader persona guidelines) and out of scope.

### CODE REVIEW

**Comment by hristog** [comment]
I'll use this PR to share some progress update as well.

I've completed end-to-end implementations, associated with the following increments:
- Recursive LCS for identification of an arbitrary longest common sequence between two given sequences.
- Dynamic Programming LCS, with a table representation, allowing recovering all longest common sequences via backtracking.
- Heuristic for identifying a diff-friendly LCS solution (amongst multiple possible LCS solutions).
- Parametrised filtering of diff results, so that it could facilitate different forms of diff formatting.
  - In this case, the implementation introduces one parameter corresponding to the maximum length of matches on either side, which should be displayed as a context in a diff solution.

**Comment by gvwilson** [comment]
I realize this is in progress, but may I merge what you have so that I have something when I start rearranging order?

**Comment by hristog** [comment]
My apologies, @gvwilson - it appears that I've misunderstood the purpose of the chapter sub-directories and the associated PR increments.

The initialisation of the sub-directory was in response to [this](https://github.com/roc-lang/book-of-examples/issues/15#issuecomment-2019121872) comment from the associated issue:
> 👍 thank you - I'll mark this one down as yours. Can you please create a sub-directory under the project root called "diff" and put your work there along with an index.md file with point-form notes as you go along?

But I wasn't sure at what frequencies should we have expected the PR increments to be merged into the repository here. I confirm I've been writing the draft of my chapter (and complete code examples) in parallel (i.e., whilst waiting for this initialisation PR to be merged).

Should I seek to update this PR with my draft work, or should that be part of a separate PR? I confirm, either way, I'll be able to provide a much better idea about the content of the chapter this weekend.

Basically (and it seems - entirely caused by my misunderstanding), I was waiting for this PR to be merged in first, so that I could subsequently raise other PRs corresponding to draft version increments.

**Comment by gvwilson** [comment]
Hi @hristog - please don't apologize, we're making this up as we go along :-) I have one more week until I start a new job, and would like to tidy up what we have before then, so I've been firing off requests for merge more hastily than I should have. You've done everything right; I'll go ahead and merge this as it is and then we can see where we are. Apologies on my part for being confusing.

**Comment by hristog** [comment]
Understood, @gvwilson! Thank you once again for confirming!
